### DISCUSSION TITLE
validation before creating a row

### DISCUSSION THREAD

**Discussion by mabdalla123** [filamentphp/filament#4135] [Help] [answered]
--in Class
// class CreateResult extends CreateRecord
--what is  the validation for 
//  check if a student  result already exists  in this semester

`protected function beforeCreate():void
    {

        $result = Result::where("semester_id",$this->data["semester_id"])
        ->where("student_id",$this->data["student_id"])->count();

            if($result !=0){
                Notification::make()
                ->title("Student result already exists  in this semester")
                ->danger()
                ->send();

                //to do (abort creation if this is true)
                 back();

            }

    }`

How can i stop  creating this record if the validation is true ?

**Comment by Z3d0X** [comment, marked as answer]
You could use a validation rule for this.
https://filamentphp.com/docs/2.x/forms/validation#unique

```php
Field::make('semester_id')->unique(callback: function (Unique $rule, Closure $get) {
    return $rule->where('student_id', $get('student_id'));
}
```

To show a notification on validation error see: https://github.com/filamentphp/filament/discussions/4081#discussioncomment-3679149

**Reply by mabdalla123** [reply]
Thanks
